Shadowling Update 2: 121% Merge-Ready

alisa-ulan opened the pull request:
This is mostly a remake of #9540, which should've probably been merged ages ago but wasn't, then should've been remade ages ago, but wasn't because Xhuis was presumably run over by an ambulance.
I shuffled a bit of the code around in the shadowling veil ability to make it more readable, and cut out some copypasted code, fixed some bugs, and made it darken monitors.

Fixes #9021, fixes #9794, fixes #9538, fixes #9197.  More generally, should fix any item ever not getting its light cut by veil, whether the item is on the ground or being worn/held/pocketed.

I'm not sure if shadowling is in the rotation, but it's even more probably 100% ready to go with these changes than ever before.

Commit:
Beginnings of new update

Commit:
More work.

Commit:
More work and moves ascendant shadowling file

Commit:
Oh I see what I did wrong.

Commit:
Removes Vortex

Commit:
Bit more work

alisa-ulan committed:
Comitterydoo

Adds Xhuis' second shadowling fix, along with a slight rework and
expansion of the veil spell.

alisa-ulan committed:
Whoops

Steelpoint commented:
This is good, I've seen some Shadowling rounds and this should go a long way in smoothing things over.

:+1:

AlexanderUlanH commented:
>Suddenly self-monkeying is the counter to sling

New commit coming right after I test it

alisa-ulan committed:
Fixes

Veil now works on non-human mobs.  There are weird edge cases with
monkeys and flashlights which exist already, but the unconditional
SetLuminosity deals with them.
Also, stops veil from assuming all flares are lit.

optimumtact commented:
@phil235 @Razharas @RemieRichards @Cheridan

Miauw62 commented on code/game/gamemodes/shadowling/ascendant_shadowling.dm:
```diff
@@ -25,9 +28,16 @@
 	return 1 //copypasta from carp code
+
+/mob/living/simple_animal/ascendant_shadowling/say(var/message)
+	message = message + "!!"
```
this should be an override of treat_message(), you should basically never override say() directly.

Miauw62 commented on code/game/gamemodes/shadowling/shadowling_abilities.dm:
```diff
@@ -142,12 +145,20 @@
+	var/thrallsPresent = 0
+	var/mob/living/carbon/human/user = usr
+	for(var/datum/mind/mindToCount in ticker.mode.thralls)
+		thrallsPresent++
```
cant you just use ticker.mode.thralls.len?

RemieRichards replied:
^ this but make sure to call listclearnulls() on ticker.mide.thralls BEFORE counting the len.
a list of 30 nulls is still 30 long!

Miauw62 commented on code/game/gamemodes/shadowling/shadowling_abilities.dm:
```diff
@@ -355,8 +370,8 @@
-/datum/reagent/shadowling_blindness_smoke //Blinds non-shadowlings, heals shadowlings/thralls
-	name = "!(%@ ERROR )!@$"
+datum/reagent/shadowling_blindness_smoke //Blinds non-shadowlings, heals shadowlings/thralls
+	name = "Odd Black Liquid"
```
should be lower case, so BYOND will automatically append "the" at the front. alternatively, prefix this with \improper, although that has been known to be buggy.

RemieRichards replied:
Don't do the "\improper" we need to deprecate that.
it causes an odd Umlaut'd Y to appear in texts and UIs.

simply lowercase the name.

Miauw62 replied:
thats what i ment by "has known to be buggy"

RemieRichards replied:
Yes but you still left it an option, I'm saying no, don't do that ever stop what are you doing no pls.

Miauw62 commented on code/game/gamemodes/shadowling/shadowling_abilities.dm:
```diff
@@ -414,72 +429,86 @@
+			targetsDrained++
+			nearbyTargets.Add(M)
+		for(var/mob/living/carbon/M in nearbyTargets)
+			nearbyTargets.Remove(M) //To prevent someone dying like a zillion times
```
is this neccesary?

AlexanderUlanH replied:
I don't see any way it could be, unless someone managed to be put in the target list twice.  Am I missing something, @Xhuis?

AlexanderUlanH replied:
It was needed because the loops are tabbed incorrectly.  Fixed now.

Miauw62 commented on code/game/gamemodes/shadowling/shadowling_abilities.dm:
```diff
@@ -414,72 +429,86 @@
+			M.adjustOxyLoss(20)
+			M.adjustStaminaLoss(20)
+			M << "<span class='boldannounce'>You feel a wave of exhaustion and a curious draining sensation directed towards [usr]!</span>"
+			usr << "<span class='shadowling'>You draw the life from [M] to heal your wounds.</span>"
```
this will probably get pretty spammy

Miauw62 commented on code/game/gamemodes/shadowling/special_shadowling_abilities.dm:
```diff
@@ -13,7 +13,7 @@
 			return
 		if("Yes")
-			usr.notransform = 1
+			usr.Stun(INFINITY) //This is bad but notransform won't work.
```
just buckle them to an invisible chair in the MC :^]

Anonus replied:
Don't do this.

AlexanderUlanH commented:
Many thanks, I'll fix those things up in a few hours.

alisa-ulan committed:
Improvements

As requested

Xhuis commented:
Thank you very much for doing this, I'm on a bit of a hiatus because of real life issues. \o/
